File: Practica2_i-Lunch/src/gestorAplicacion/gestionRestaurante/pedido.py
class Pedido:

    # Atributos estaticos

    _pedidos = []
    _totalPedidos = 0

    # Constructor

    def __init__(self, cliente = None, tipo = "", estado = "", mensaje = "", fechaHora = None, precioTotal = 0, restaurante = None):
        self._cliente = cliente
        self._tipo = tipo
        self._estado = estado
        self._mensaje = mensaje
        self._fechaHora = fechaHora
        self._precioTotal = precioTotal
        self._restaurante = restaurante
        
        self._codigo = Pedido._totalPedidos
        self._productos = []
        cliente.agregarPedidoHistorial(self)
        Pedido._pedidos.append(self)
        Pedido._totalPedidos += 1

    # Getters y Setters

    def getCliente(self):
        return self._cliente

    def setCliente(self, nuevoCliente):
        self._cliente = nuevoCliente

    def getCodigo(self):
        return self._codigo

    def setCodigo(self, codigo):
        self._codigo = codigo

    def getEstado(self):
        return self._estado

    def setEstado(self, estado):
        self._estado = estado

    def getTipo(self):
        return  self._tipo

    def setTipo(self, tipo):
        self._tipo = tipo

    def getMensaje(self):
        return self._mensaje

    def setMensaje(self, mensaje):
        self._mensaje = mensaje

    def getFechaHora(self):
        return self._fechaHora

    def setFechaHora(self, fechaHora):
        self._fechaHora = fechaHora

    def getPrecioTotal(self):
        return self._precioTotal

    def setPrecioTotal(self, precioTotal):
        self._precioTotal = precioTotal

    def getRestaurante(self):
        return self._restaurante

    def setRestaurante(self, restaurante):
        self._restaurante = restaurante

    def getProductos(self):
        return self._productos

    def setProductos(self, productos):
        self._productos = productos

    @classmethod
    def getPedidos(cls):
        return cls._pedidos

    @classmethod
    def setPedidos(cls, pedidos):
        cls._pedidos = pedidos

    @classmethod
    def getTotalPedidos(cls):
        return cls._totalPedidos

    @classmethod
    def setTotalPedidos(cls, totalPedidos):
        cls._totalPedidos = totalPedidos

    # Metodos
    
    def calcularPrecioTotal(self):
        sum = 0
        for i in self._productos:
            sum += i.getPrecio()
            
        return sum

    # ToString

    def __str__(self):
        return f"Pedido {self._codigo} hecho por el cliente {self._cliente.getNombre()}\n" \
               f"Fecha y hora: {self._fechaHora}\n" \
               f"Estado: {self._estado}\n" \
               f"Mensaje: {self._mensaje}\n" \
               f"Tipo: {self._tipo}\n" \
               f"Precio total: ${self._precioTotal}"

File: Practica1_i-Lunch/src/gestorAplicacion/gestionRestaurante/Pedido.java
package gestorAplicacion.gestionRestaurante;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.ArrayList;

import gestorAplicacion.usuariosRestaurante.*;

public class Pedido implements Serializable {

	// Serializacion
	private static final long serialVersionUID = 1L;
	private static ArrayList<Pedido> pedidos;
	static {
		pedidos = new ArrayList<Pedido>();
	}

	////////////// ATRIBUTOS //////////////
	private static int totalPedidos = 0;
	private Cliente cliente;
	private int codigo;
	private String estado;

	// ArrayList donde se encuentran todos los productos que componen el pedido.
	private ArrayList<Producto> productos = new ArrayList<Producto>();

	private String tipo;
	// Pienso que es mejor separar fecha.
	private LocalDateTime fechaHora;

	// Mensaje agregado por el usuario de forma opcional al realizar un pedido.
	private String mensaje;
	private int precioTotal;
	private Restaurante restaurante;

	////////////// METODOS ////////////////

	public int calcularPrecioTotal() {
		int sum = 0;
		for (int i = 0; i < productos.size(); i++) {
			sum += productos.get(i).getPrecio();
		}
		return sum;
	}

	/////////// CONSTRUCTORES /////////////

	public Pedido(Cliente cliente, int codigo, String tipo, LocalDateTime fechaHora, Restaurante restaurante) {
		// Considero que estos son los datos necesarios para empezar un pedido
		this.cliente = cliente;
		this.codigo = codigo;
		this.tipo = tipo;
		this.fechaHora = fechaHora;
		this.restaurante = restaurante;
		pedidos.add(this);
		Pedido.totalPedidos += 1;

		// Sumar a los pedidos del cliente
		this.cliente.agregarPedidoHistorial(this);
	}

	public Pedido() {
		this(null, 0, "", null, null);
	}

	public Pedido(Cliente cliente) {
		this(cliente, 0, "", null, null);
	}

	///////// GETTERS AND SETTERS /////////

	public Cliente getCliente() {
		return cliente;
	}

	public void setCliente(Cliente cliente) {
		this.cliente = cliente;
	}

	public int getCodigo() {
		return codigo;
	}

	public void setCodigo(int codigo) {
		this.codigo = codigo;
	}

	public String getEstado() {
		return estado;
	}

	public void setEstado(String estado) {
		this.estado = estado;
	}

	public ArrayList<Producto> getProductos() {
		return productos;
	}

	public void setProductos(ArrayList<Producto> productos) {
		this.productos = productos;
	}

	public String getTipo() {
		return tipo;
	}

	public void setTipo(String tipo) {
		this.tipo = tipo;
	}

	public LocalDateTime getFechaHora() {
		return fechaHora;
	}

	public void setFechaHora(LocalDateTime fechaHora) {
		this.fechaHora = fechaHora;
	}

	public String getMensaje() {
		return mensaje;
	}

	public void setMensaje(String mensaje) {
		this.mensaje = mensaje;
	}

	public int getPrecioTotal() {
		return precioTotal;
	}

	public void setPrecioTotal(int precioTotal) {
		this.precioTotal = precioTotal;
	}

	public Restaurante getRestaurante() {
		return restaurante;
	}

	public void setRestaurante(Restaurante restaurante) {
		this.restaurante = restaurante;
	}

	public static ArrayList<Pedido> getPedidos() {
		return pedidos;
	}

	public static int getTotalPedidos() {
		return pedidos.size();
	}

	public static void setPedidos(ArrayList<Pedido> pedidos) {
		Pedido.pedidos = pedidos;
	}

	public static void setTotalPedidos(int totalPedidos) {
		Pedido.totalPedidos = totalPedidos;
	}

	@Override
	public String toString() {
		String productosString = "";

		// Fecha hora
		DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd - HH:mm:ss");
		String fechaHoraString = fechaHora.format(formatter);

		for (Producto producto : this.productos) {
			productosString += " - " + producto.getNombre() + " (" + producto.getCantidad() + ")\n";
		}

		return "\nPedido " + codigo + " hecho por el cliente: \"" + cliente.getNombre() + "\"\n" + "Fecha y hora: "
				+ fechaHoraString + "\n" + "Estado: " + estado + "\n" + "Mensaje: " + mensaje + "\n" + "Tipo: " + tipo
				+ "\n" + "Productos: \n" + productosString + "Precio total: $" + precioTotal;
	}
}
